CoreFoundation/Parsing.subproj/CFXMLInterface.c

```diff
@@ -216,4 +233,386 @@ void _CFXMLInterfaceSAX2UnparsedEntityDecl(_CFXMLInterfaceParserContext ctx, con
+    return xmlNewProp(node, name, value);
+}
+
+_CFXMLNamespacePtr _CFXMLNewNamespace(_CFXMLNodePtr node, const unsigned char* uri, const unsigned char* prefix) {
+    return xmlNewNs(node, uri, prefix);
+}
+
+CF_RETURNS_RETAINED CFStringRef _CFXMLNodeURI(_CFXMLNodePtr node) {
+    xmlNodePtr nodePtr = (xmlNodePtr)node;
+    switch (nodePtr->type) {
+        case XML_ATTRIBUTE_NODE:
+        case XML_ELEMENT_NODE:
+            return CFStringCreateWithCString(NULL, (const char*)nodePtr->ns->href, kCFStringEncodingUTF8);
+
+        case XML_DOCUMENT_NODE:
+        {
+            xmlDocPtr doc = (xmlDocPtr)node;
+            return CFStringCreateWithCString(NULL, (const char*)doc->URL, kCFStringEncodingUTF8);
+        }
+
+        default:
+            return NULL;
+    }
+}
+
+void _CFXMLNodeSetURI(_CFXMLNodePtr node, const unsigned char* URI) {
+    xmlNodePtr nodePtr = (xmlNodePtr)node;
+    switch (nodePtr->type) {
+        case XML_ATTRIBUTE_NODE:
+        case XML_ELEMENT_NODE:
+
+            if (!URI) {
+                if (nodePtr->ns) {
+                    xmlFree(nodePtr->ns);
+                }
+                nodePtr->ns = NULL;
+                return;
+            }
+
+            xmlNsPtr ns = xmlSearchNsByHref(nodePtr->doc, nodePtr, URI);
+            if (!ns) {
+                if (nodePtr->ns && (nodePtr->ns->href == NULL)) {
+                    nodePtr->ns->href = xmlStrdup(URI);
+                    return;
+                }
+
+                ns = xmlNewNs(nodePtr, URI, NULL);
+            }
+
+            xmlSetNs(nodePtr, ns);
+            break;
+
+        case XML_DOCUMENT_NODE:
+        {
+            xmlDocPtr doc = (xmlDocPtr)node;
+            if (doc->URL) {
+                xmlFree((xmlChar*)doc->URL);
+            }
+            doc->URL = URI;
+        }
+            break;
+
+        default:
+            return;
+    }
+}
+
+void _CFXMLNodeSetPrivateData(_CFXMLNodePtr node, void* data) {
+    ((xmlNodePtr)node)->_private = data;
+}
+
+void* _Nullable  _CFXMLNodeGetPrivateData(_CFXMLNodePtr node) {
+    return ((xmlNodePtr)node)->_private;
+}
+
+_CFXMLNodePtr _CFXMLNodeProperties(_CFXMLNodePtr node) {
+    return ((xmlNodePtr)node)->properties;
+}
+
+CFIndex _CFXMLNodeGetType(_CFXMLNodePtr node) {
+    return ((xmlNodePtr)node)->type;
+}
+
+const char* _CFXMLNodeGetName(_CFXMLNodePtr node) {
+    return (const char*)(((xmlNodePtr)node)->name);
+}
+
+void _CFXMLNodeSetName(_CFXMLNodePtr node, const char* name) {
+    xmlNodeSetName(node, (const xmlChar*)name);
+}
+
+CFStringRef _CFXMLNodeGetContent(_CFXMLNodePtr node) {
+    xmlChar* content = xmlNodeGetContent(node);
+    if (content == NULL) {
+        return NULL;
+    }
+    CFStringRef result = CFStringCreateWithCString(NULL, (const char*)content, kCFStringEncodingUTF8);
+    xmlFree(content);
+
+    return result;
+}
+
+const char* _getCStringFromCFString(CFStringRef string);
```

Comment:
Whoops! That particular last line isn't used, it didn't get deleted when I realized I didn't need such a method. I'll clean that up as soon as I get to a computer.
Thanks,Robert Thompson

Sent from Outlook on my iPhone




On Thu, Jan 7, 2016 at 4:11 PM -0800, "Philippe Hausler" <notifications@github.com> wrote:












In CoreFoundation/Parsing.subproj/CFXMLInterface.c:
> +void _CFXMLNodeSetName(_CFXMLNodePtr node, const char* name) {
> +    xmlNodeSetName(node, (const xmlChar*)name);
> +}
> +
> +CFStringRef _CFXMLNodeGetContent(_CFXMLNodePtr node) {
> +    xmlChar* content = xmlNodeGetContent(node);
> +    if (content == NULL) {
> +        return NULL;
> +    }
> +    CFStringRef result = CFStringCreateWithCString(NULL, (const char*)content, kCFStringEncodingUTF8);
> +    xmlFree(content);
> +
> +    return result;
> +}
> +
> +const char* _getCStringFromCFString(CFStringRef string);



where is this used in here?



—
Reply to this email directly or view it on GitHub.